ansible_base/lib/utils/models.py

```diff
@@ -97,3 +99,101 @@ def current_user_or_system_user():
     if user is None or user.is_anonymous:
         user = get_system_user()
     return user
+
+
+def diff(old, new, require_type_match=True, json_safe=True, include_m2m=False, exclude_fields=[], limit_fields=[]):
+    """
+    Diff two instances of models (which do not have to be the same type of model
+    if given require_type_match=False).
+
+    This function is used in particular by the activitystream application where
+    the changes returned by this function are stored as models change.
+
+    :param old: The old instance for comparison
+    :param new: The new instance for comparison
+    :param require_type_match: If True, old and new must be of the same type of
+        model. (default: True)
+    :param json_safe: If True, the diff will be made JSON-safe by converting
+        all non-JSON-safe values to strings using Django's smart_str function.
+        (default: True)
+    :param exclude_fields: A list of field names to exclude from the diff.
+        (default: [])
```

Comment:
I can't find uses in the actual functional code that use these options.

Reply:
Yes, and that's a bug, good find. The class variable `activity_stream_excluded_field_names` should work, but we never pass that to this function from the signals.